src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,145 @@
+import { Observable } from "rxjs/Observable";
+import compareVersions from "compare-versions";
+import { request, RequestResponse } from "@dcos/http-service";
+import { BehaviorSubject } from "rxjs";
+
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+export const UpdateStreamType: symbol = Symbol("UpdateStreamType");
+const CHECK_DELAY: number = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION_SETTING: string = "dcosUIVersion";
+const DISMISSED_VERSION: string = "dismissedVersion";
+const LAST_TIME_CHECK: string = "lastTimeCheck";
+
+interface ContentTypeObject {
+  action: string;
+  actionType: string;
+  entity: string;
+  version: string;
+}
+
+interface DCOSUserSettings {
+  dcosUIVersion: DCOSUIVersion;
+  [key: string]: DCOSUIVersion;
+}
+
+interface DCOSUIVersion {
+  dismissedVersion: string;
+  lastTimeCheck: number;
+  [key: string]: string | number;
+}
+
+export function showNotification(newVersion: string): void {
+  // TODO: Move inside UpdateServiceBanner
+  alert(newVersion);
```

Comment:
This alert seems to be wrong here

Reply:
So we can't really merge this PR until mikes PR is done? We should note is as a dependency

Reply:
It's the other way around. Mike uses this PR to make his stuff work.

Reply:
It can still be reviewed separately, it just won't have a visible impact in the UI.